tools/fromschema.py

```diff
@@ -448,20 +442,15 @@ def generate_footer(schema):
         if key == 'see_also':
             # Wrap see_also list with comma separated values
             output(esc_underscores(', '.join(schema[key])))
-        elif key.startswith('example_json_') and len(schema[key]) > 0:
-            # For printing example_json_response and example_json_notifications into code block
+        elif key == 'json_examples' and len(schema[key]) > 0:
             for i, example in enumerate(schema.get(key, [])):
-                # For printing string elements from example json response; example: `createonion`
-                if isinstance(example, str):
-                    output(example)
-                    if i + 1 < len(schema[key]):
-                        output('\n')
-                else:
-                    if i == 0:
-                        output('```json\n')
-                    output(json.dumps(example, indent=2).strip() + '\n')
-                    if i + 1 == len(schema[key]):
-                        output('```')
+                output('{}Example {}:\n'.format('' if i == 0 else '\n', i + 1))
+                output('```json\n')
+                output('Request:\n')
+                output(json.dumps(example['request'], indent=2).strip() + '\n')
+                output('\nResponse:\n')
+                output(json.dumps(example['response'], indent=2).strip() + '\n')
+                output('```')
```

Comment:
This does not actually make valid JSON... you probably want the the `Request:` and `Response:` headers outside the "```json:"?


Might also be worth allowing the schema to define an optional name/descriptive line for the  example?  Not worth populating now, but might be good in future for more complex examples?

Reply:
> This does not actually make valid JSON... you probably want the the `Request:` and `Response:` headers outside the "```json:"?
 
I realised that before but wanted to keep request and response in the same block. I adjusted it according to your advice now.

> Might also be worth allowing the schema to define an optional name/descriptive line for the example? Not worth populating now, but might be good in future for more complex examples?

Also added a `description` array which can be used to explain the example in detail. Included a simple description for `dev-forget-channel' examples for demonstration.